fail2ban/client/fail2banclient.py

```diff
@@ -0,0 +1,462 @@
+			if self._conf.get("interactive", False):
+				output('  ## start ... ')
+			return self.__processCommand(['start'])
+
+		elif len(cmd) >= 1 and cmd[0] == "reload":
+			if self.__ping():
+				if len(cmd) == 1:
+					jail = 'all'
+					ret, stream = self.readConfig()
+				else:
+					jail = cmd[1]
+					ret, stream = self.readConfig(jail)
+				# Do not continue if configuration is not 100% valid
+				if not ret:
+					return False
+				self.__processCmd([['stop', jail]], False)
+				# Configure the server
+				return self.__processCmd(stream, True)
+			else:
+				logSys.error("Could not find server")
+				return False
+
+		else:
+			return self.__processCmd([cmd])
+
+
+	def __processStartStreamAfterWait(self, *args):
+		try:
+			# Wait for the server to start
+			if not self.__waitOnServer(): # pragma: no cover
+				logSys.error("Could not find server, waiting failed")
+				return False
+				# Configure the server
+			self.__processCmd(*args)
+		except ServerExecutionException as e: # pragma: no cover
+			if self._conf["verbose"] > 1:
+				logSys.exception(e)
+			logSys.error("Could not start server. Maybe an old "
+						 "socket file is still present. Try to "
+						 "remove " + self._conf["socket"] + ". If "
+						 "you used fail2ban-client to start the "
+						 "server, adding the -x option will do it")
+			if self._server:
+				self._server.quit()
+			return False
+		return True
+
+	def __waitOnServer(self, alive=True, maxtime=None):
+		if maxtime is None:
+			maxtime = self._conf["timeout"]
+		# Wait for the server to start (the server has 30 seconds to answer ping)
+		starttime = time.time()
+		logSys.debug("__waitOnServer: %r", (alive, maxtime))
+		test = lambda: os.path.exists(self._conf["socket"]) and self.__ping() 
+		with VisualWait(self._conf["verbose"]) as vis:
+			sltime = 0.0125 / 2
+			while self._alive:
+				runf = test()
+				if runf == alive:
+					return True
+				now = time.time()
+				# Wonderful visual :)
+				if now > starttime + 1:
+					vis.heartbeat()
+				# f end time reached:
+				if now - starttime >= maxtime:
+					raise ServerExecutionException("Failed to start server")
+				sltime = min(sltime * 2, 0.5)
+				time.sleep(sltime)
+		return False
+
+	def start(self, argv):
+		# Install signal handlers
+		_prev_signals = {}
+		if _thread_name() == '_MainThread':
+			for s in (signal.SIGTERM, signal.SIGINT):
+				_prev_signals[s] = signal.getsignal(s)
+				signal.signal(s, self.__sigTERMhandler)
+		try:
+			# Command line options
+			if self._argv is None:
+				ret = self.initCmdLine(argv)
+				if ret is not None:
+					if ret:
+						return True
+					raise ServerExecutionException("Init of command line failed")
+
+			# Commands
+			args = self._args
+
+			# Interactive mode
+			if self._conf.get("interactive", False):
+				# no readline in test:
+				if PRODUCTION: # pragma: no cover
+					try:
+						import readline
+					except ImportError:
+						raise ServerExecutionException("Readline not available")
```

Comment:
hm... I have removed all PRODUCTION readline conditioning in this file and tests still passed no problem (Python 2.7.11+ , Debian system)... what is the catch? ;)